src/Plugin/Field/FieldWidget/InlineEntityFormMultiple.php

```diff
@@ -240,7 +240,6 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
     $element['#element_validate'] = array('inline_entity_form_update_row_weights');
     // Add the required element marker & validation.
     if ($element['#required']) {
-      $element['#title'] .= ' ' . _theme('form_required_marker', array('element' => $element));
```

Comment:
I researched this a bit.... fieldset will respect #required and add star to the title appropriately. Problem here is that we actually have a fieldset inside fieldset (one is $element and the other is $element['form']) and it looks that the second is the one that is actually displayed. Taking this into consideration it seems that adding correct values for #title and #required on the second fieldset should do what we want here.